daffodil-japi/build.sbt

```diff
@@ -18,7 +18,7 @@
 enablePlugins(GenJavadocPlugin)
 enablePlugins(PublishJavadocPlugin)
 
-unidocGenjavadocVersion := "0.18_2.12.15"
+unidocGenjavadocVersion := "0.18_2.13.8"
```

Comment:
I think we don't actually want to merge this? The CI tests pass, but I assume we want to stick with the version of genjavadoc that matches our scala version (2.12.15)?

In fact, looking at the GenJavaDocPlugin, I wonder if we should even being including the scala version in the value of `unidocGenjavadocVersion`? Looking at this file:

https://github.com/sbt/sbt-unidoc/blob/master/src/main/scala/sbtunidoc/GenJavadocPlugin.scala

The setting defaults to just "0.18" and uses `crossVersion` to get the right Scala version. Seems like maybe we should change this  back to just "0.18"?

Reply:
I looked at <https://github.com/sbt/sbt-unidoc/issues/110>, <https://github.com/sbt/sbt-unidoc>, and <https://github.com/apache/daffodil/blame/main/daffodil-japi/build.sbt> to determine why we need that unidocGenjavadocVersion line (we need it because the old default, 0.15, doesn't work anymore, a PR changed it to 0.18 in the sbt-unidoc repo only yesterday, and a new sbt-unidoc release hasn't been made yet), whether we should keep that unidocGenjavadocVersion line (yes because the default in the last released sbt-unidoc version is still 0.15), when we first added that unidocGenjavadocVersion line (in your PR <https://github.com/apache/daffodil/pull/398> almost two years ago), and why it changed to include a Scala cross version (blame Scala Steward for adding it in PR <https://github.com/apache/daffodil/pull/634>).  I agree you can change it back to "0.18" and leave it there, but Scala Stewart may add a new Scala cross version to it again in the future.  If you change that line anyway, you probably want to add a comment to remind us that the cross version shouldn't be needed and find a way to tell Scala Stewart not to add a cross version again (if possible).

Reply:
Thanks for confirming my thinking. Looking at the scala-steward configuration options, nothing jumps out at me as a way to prevent this. And we don't want to ignore this line altogether since we'll still want PRs if genjavadoc creates a new release.

I've fixed this and added a comment so we hopefully catch it if Scala Steward tries this again.